mimic/model/flavor_objects.py

```diff
@@ -1,279 +1,107 @@
-                                           "rxtx_factor": 800, "disk": 80},
-               "8 GB General Purpose v1": {"id": "general1-8", "ram": 8192, "vcpus": 8,
-                                           "rxtx_factor": 1600, "disk": 160}}
-
-    def extra_specs_json(self):
-        """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for a general purpose flavor.
-        """
-        return {
-            "class": "general1",
-            "policy_class": "general_flavor"
-        }
 
-
-class RackspaceIOFlavor(Flavor):
+@attributes(
+    ["tenant_id", "region_name", "clock",
+     Attribute("flavors_store", default_factory=list)]
+)
+class RegionalFlavorCollection(object):
     """
-    A Rackspace IO flavor object representation
+    A collection of flavors, in a given region, for a given tenant.
     """
-    flavors = {"15 GB I/O v1": {"id": "io1-15", "ram": 15360, "vcpus": 4, "rxtx_factor": 1250,
-                                "disk": 40},
-               "30 GB I/O v1": {"id": "io1-30", "ram": 30720, "vcpus": 8, "rxtx_factor": 2500,
-                                "disk": 40},
-               "60 GB I/O v1": {"id": "io1-60", "ram": 61440, "vcpus": 16, "rxtx_factor": 5000,
-                                "disk": 40},
-               "90 GB I/O v1": {"id": "io1-90", "ram": 92160, "vcpus": 24, "rxtx_factor": 7500,
-                                "disk": 40},
-               "120 GB I/O v1": {"id": "io1-120", "ram": 122880, "vcpus": 32, "rxtx_factor": 10000,
-                                 "disk": 40}}
-
-    def extra_specs_json(self):
+    def flavor_by_id(self, flavor_id):
         """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for a memory flavor.
+        Retrieve a :obj:`Flavor` object by its ID.
         """
-        return {
-            "class": "io1",
-            "policy_class": "io_flavor"
-        }
-
-
-class RackspaceMemoryFlavor(Flavor):
-    """
-    A Rackspace memory flavor object representation
-    """
-    flavors = {"15 GB Memory v1": {"id": "memory1-15", "ram": 15360, "vcpus": 2, "rxtx_factor": 625,
-                                   "disk": 0},
-               "30 GB Memory v1": {"id": "memory1-30", "ram": 30720, "vcpus": 4, "rxtx_factor": 1250,
-                                   "disk": 0},
-               "60 GB Memory v1": {"id": "memory1-60", "ram": 61440, "vcpus": 8, "rxtx_factor": 2500,
-                                   "disk": 0},
-               "120 GB Memory v1": {"id": "memory1-120", "ram": 122880, "vcpus": 16, "rxtx_factor": 5000,
-                                    "disk": 0},
-               "240 GB Memory v1": {"id": "memory1-240", "ram": 245760, "vcpus": 32,
-                                    "rxtx_factor": 10000, "disk": 0}}
+        if len(self.flavors_store) < 1:
+            self.create_flavors_list()
```

Comment:
non-blocking: I see this needs to be lazy loaded for some reason, and I see looking down at `create_flavors_list` it also depends on the tenant ID. Really what would be nice here is to have the attribute initializer set all this up, but `characteristic` isn't sophisticated enough to do that automatically. I wonder if `attrs` could configure it in such a way?